src/ArmTemplates/Extractor/EntityExtractors/APIExtractor.cs

```diff
@@ -279,5 +280,32 @@ public async Task<ApiTemplateResources> GetApiRelatedTemplateResourcesAsync(stri
 
             return apiDependency;
         }
+
+        void ValidateOutputApiTemplateResource(string apiName, ApiTemplateResource apiResource, ApiTemplateResources apiTemplateResources)
+        {
+            //remove api operation if websocket type & change dependencies to api, since api operation for websocket type is created automatically and it is not allowed to update it (built-in operation)
+            if (apiResource.Properties.Type == ApiTypeConstants.WebSocket)
+            {
+                if (!apiTemplateResources.ApiOperations.IsNullOrEmpty())
+                {
+                    if (!apiTemplateResources.ApiOperationsPolicies.IsNullOrEmpty())
+                    {
+                        foreach (var apiOperationPolicy in apiTemplateResources.ApiOperationsPolicies)
+                        {
+                            apiOperationPolicy.DependsOn = new string[] { NamingHelper.GenerateApisResourceId(apiName) };
+                        }
+                    }
+
+                    if (!apiTemplateResources.ApiOperationsTags.IsNullOrEmpty())
+                    {
+                        foreach (var apiOperationTag in apiTemplateResources.ApiOperationsTags)
+                        {
+                            apiOperationTag.DependsOn = new string[] { NamingHelper.GenerateApisResourceId(apiName) };
+                        }
+                    }
+                    apiTemplateResources.ApiOperations.Clear();
```

Comment:
it can clear not only WebSocket apis. Is that expected? Or I miss anything?

Reply:
It is not expected, since the apiTemplateResources contains only information about current Api which being processed, called from the method: https://github.com/Azure/azure-api-management-devops-resource-kit/blob/057d815968dded485c25d466a122903bd35fc746/src/ArmTemplates/Extractor/EntityExtractors/APIExtractor.cs#L100